lib/src/ardrive_io.dart

```diff
@@ -9,6 +9,18 @@ import 'mobile/mobile_io.dart';
 import 'web/stub_web_io.dart' // Stub implementation
     if (dart.library.html) 'web/web_io.dart';
 
+class SaveStatus {
+  final int bytesSaved;
+  final int totalBytes;
+  final bool? saveResult;
```

Comment:
~Does the null value have a specific meaning here? Is it different than defaulting to false?~

**Edit -** Having it be null means the download hasn't yet finished.

Reply:
We can rename it to `hasFinished` or `isResultSaved`, or similar.